File: CHAP -- @NUMERIC PROBLEM/CHAP -- BlueJCodes/SourceCode-JCreatorPro/PracticeCode/src/ThisKeyword/ThisCode.java
/* this: to refer current class instance variable */
import java.util.*;

class Demo1
{
    int age;
    String name;

    public Demo1(int age, String name)
    {
        this.age = age;
        this.name = name;
    }

    void Show()
    {
        System.out.println();
        System.out.println("Your age is: " + Integer.toString(age));
        System.out.println("Your name is: " + name);
    }
}

class ThisCode
{
    public static void main(String[] args)
    {
        int _age;
        String _name;

		Scanner in = new Scanner(System.in);
        System.out.println("Enter your age: ");
        _age = in.nextInt();

        System.out.println("Enter your name: ");
        _name = in.next();

        Demo1 obj = new Demo1(_age, _name);

        obj.Show();
        System.out.println();
        System.out.println("Please any key to continue...!");
    }
}

File: CHAP -- @Notes/java/RecapsulationCodes/RecapsulationCodes/ThisKeyword/ThisCode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace RecapsulationCodes.ThisKeyword
{
    class Demo1
    {
        int age;
        string name;

        public Demo1(int age, string name)
        {
            this.age = age;
            this.name = name;
        }

        public void Show()
        {
            Console.WriteLine();
            Console.WriteLine("Your age is: " + age.ToString());
            Console.WriteLine("Your name is: " + name);
        }
    }


    class ThisCode
    {
        static void Main(string[] args)
        {
            int _age;
            string _name;

            Console.Write("Enter your age: ");
            _age = Int32.Parse(Console.ReadLine());

            Console.Write("Enter your name: ");
            _name = Console.ReadLine();

            Demo1 obj = new Demo1(_age, _name);
            obj.Show();
            Console.WriteLine();
            Console.WriteLine("Please any key to continue...!");
            Console.ReadKey(true);
        }
    }
}
